### PR TITLE
update to latest netcode

### PR DESCRIPTION
**Pull request by SamuelBellomo**
We still need to do rebranding for boss room itself, this is just to adapt to netcode sdk's own rebranding

### CODE REVIEW

File: ProjectSettings/ProjectVersion.txt
Comment (LukeStampfli) [inline ProjectSettings/ProjectVersion.txt:1]: Is this wanted or here by mistake?

**Review by LukeStampfli — APPROVED** [review]
Namespace changes look all good.

Two things to point out:
- PR title includes NGO. We don't use NGO here 😃. Use "Netcode" instead.
- I saw a lot of unused namespaces being removed and full type specifiers such as "MLAPI.Transports.UNET.UNetTransport" etc. Maybe consider something similar to what the SDK repo is doing where they have a 'standards.py' and use dotnet format.

File: ProjectSettings/ProjectVersion.txt
Comment (fernando-cortez) [reply ProjectSettings/ProjectVersion.txt:1]: Bumping this. Was a version update necessary?

**Review by fernando-cortez — COMMENTED** [review]
Besides the version comment, all looks good to me. Will approve once that's addressed.

File: ProjectSettings/ProjectVersion.txt
Comment (Cosmin-B) [reply ProjectSettings/ProjectVersion.txt:1]: For UTP + Relay integration, I bumped this up to .12f1 but I don't think is necessary

**Comment by unity-cla-assistant** [comment]
[CLA assistant check](https://cla-assistant.cds.internal.unity3d.com/Unity-Technologies/com.unity.multiplayer.samples.coop?pullRequest=359) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.cds.internal.unity3d.com/Unity-Technologies/com.unity.multiplayer.samples.coop?pullRequest=359) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.cds.internal.unity3d.com/check/Unity-Technologies/com.unity.multiplayer.samples.coop?pullRequest=359) it.</sub>

**Review by fernando-cortez — COMMENTED** [review]
Tested and looks good to go. Just needs conflicts to be resolved. @VALERE91 or @SamuelBellomo

**Comment by LukeStampfli** [comment]
I updated the branch to the new transport interface (no more channels) and we are now using the develop version of the litenetlib and photon transport temporarily until the contribution repository content gets updated.

**Review by LukeStampfli — APPROVED** [review]

**Review by Cosmin-B — APPROVED** [review]
UTP merge is going to follow

File: Packages/manifest.json
Comment (Cosmin-B) [inline Packages/manifest.json:14]: I don't like that this is here, as this repo is still private + is using git ssh instead of https

**Comment by Cosmin-B** [comment]
This commit, corrupted BossRoom and CharacterSelect scenes, they weren't merged properly! https://github.com/Unity-Technologies/com.unity.multiplayer.samples.coop/pull/359/commits/55a7c9a838c0b4aca5de50f4739e6c0fbc724c33

image

**Review by Cosmin-B — CHANGES_REQUESTED** [review]
See my last commit

**Comment by LukeStampfli** [comment]
It works on my machine

**Comment by Cosmin-B** [comment]
> It works on my machine

I will nuke everything, and pull it again and try it, let me see

**Review by Cosmin-B — APPROVED** [review]

File: Packages/manifest.json
Comment (SamuelBellomo) [reply Packages/manifest.json:14]: @VALERE91Unity I understand you used this for profiling. Do we still need this or can we add it later once it's available publicly?

Comment (VALERE91Unity) [reply Packages/manifest.json:14]: Technically yes let me try to be 100% sure and if yes I'm gonna push without it

**Comment by VALERE91Unity** [comment]
> This commit, corrupted BossRoom and CharacterSelect scenes, they weren't merged properly! [55a7c9a](https://github.com/Unity-Technologies/com.unity.multiplayer.samples.coop/commit/55a7c9a838c0b4aca5de50f4739e6c0fbc724c33)
> 
> image

Yeah I had this problem a lot during the merge. It seems to be a LFS not updating properly sometime. I fixed it by doing a LFS pull.

File: Assets/BossRoom/Scripts/Shared/Net/NetworkVariableLobbyState.cs
Comment (SamuelBellomo) [inline Assets/BossRoom/Scripts/Shared/Net/NetworkVariableLobbyState.cs:12]: why not use a state like NetworkTransform is using? this would simplify your serialization code. You could still have a separate lobby state class.

File: Assets/BossRoom/Scripts/Shared/Net/NetworkVariableGUID.cs
Comment (SamuelBellomo) [inline Assets/BossRoom/Scripts/Shared/Net/NetworkVariableGUID.cs:9]: same here, why do we need a whole new network variable for this?

File: Assets/BossRoom/Scripts/Shared/Net/NetworkVariableLobbyState.cs
Comment (SamuelBellomo) [reply Assets/BossRoom/Scripts/Shared/Net/NetworkVariableLobbyState.cs:12]: @VALERE91Unity

File: Assets/BossRoom/Scripts/Shared/Net/NetworkVariableGUID.cs
Comment (SamuelBellomo) [reply Assets/BossRoom/Scripts/Shared/Net/NetworkVariableGUID.cs:9]: @VALERE91Unity

Comment (VALERE91Unity) [reply Assets/BossRoom/Scripts/Shared/Net/NetworkVariableGUID.cs:9]: A GUID is managed so we cannot put it in a NetworkVariable<>

File: Assets/BossRoom/Scripts/Shared/Net/NetworkVariableLobbyState.cs
Comment (VALERE91Unity) [reply Assets/BossRoom/Scripts/Shared/Net/NetworkVariableLobbyState.cs:12]: I think it was because of the absolute need to be unmanaged. And once something is managed in a struct (event an array or a string) the whole struct becomes managed and cannot fit in a NetVariable anymore.

File: Assets/BossRoom/Scripts/Shared/Net/NetworkVariableGUID.cs
Comment (LukeStampfli) [reply Assets/BossRoom/Scripts/Shared/Net/NetworkVariableGUID.cs:9]: There is an alternative way of implementing this. Createa struct `NetworkGUID` have it store the guid as 2 ulongs. That way it's an unmanaged struct. Have it implement INetworkSerializable and then put it into `NetworkVariable<T>` that's probably a bit cleaner than a custom netvar.

Comment (VALERE91Unity) [reply Assets/BossRoom/Scripts/Shared/Net/NetworkVariableGUID.cs]: I change it for the struct version :)

**Comment by SamuelBellomo** [comment]
looks good to me 🚀 
if I can get two approvals, we'll be able to merge this

**Review by fernando-cortez — APPROVED** [review]

**Pull request merged**
